lib/reek/report/report.rb

```diff
@@ -144,53 +144,46 @@ class XMLReport < Base
       require 'rexml/document'
 
       def show
-        checkstyle = REXML::Element.new('checkstyle', document)
-
-        smells.group_by(&:source).each do |file, file_smells|
-          file_to_xml(file, file_smells, checkstyle)
-        end
-
-        print_xml(checkstyle.parent)
+        document.write $stdout, 2
```

Comment:
1. `document.write output: $stdout, indent: 2`
2. `REXML::Formatters::Pretty.new.write document, $stdout`

Either if fine by me; the former is shorter and more readable, neither is very object-oriented (document shouldn’t really write itself, formatter should just format and return the formatted version, not write…).